File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/android/src/main/java/im/zego/zego_express_engine/ZegoCustomAudioCaptureManager.java
package im.zego.zego_express_engine;

import java.nio.ByteBuffer;

import im.zego.zegoexpress.ZegoExpressEngine;
import im.zego.zegoexpress.entity.ZegoAudioFrameParam;
import im.zego.zegoexpress.constants.ZegoAudioSampleRate;
import im.zego.zegoexpress.constants.ZegoAudioChannel;
import im.zego.zegoexpress.constants.ZegoPublishChannel;

public class ZegoCustomAudioCaptureManager {
    private volatile static ZegoCustomAudioCaptureManager singleton;

    public ZegoCustomAudioCaptureManager() {
    }

    public static synchronized ZegoCustomAudioCaptureManager getInstance() {
        if (singleton == null) {
            singleton = new ZegoCustomAudioCaptureManager();
        }
        return singleton;
    }


    public void sendCustomAudioCapturePCMData(ByteBuffer data, int dataLength,
                                              ZGFlutterAudioFrameParam param,
                                              ZGFlutterPublishChannel channel) {
        ZegoAudioFrameParam audioFrameParam = new ZegoAudioFrameParam();
        audioFrameParam.sampleRate = ZegoAudioSampleRate.getZegoAudioSampleRate(param.sampleRate.value());
        audioFrameParam.channel = ZegoAudioChannel.getZegoAudioChannel(param.channel.value());
        if (ZegoExpressEngine.getEngine() != null) {
            ZegoExpressEngine.getEngine().sendCustomAudioCapturePCMData(data, dataLength, audioFrameParam, ZegoPublishChannel.getZegoPublishChannel(channel.value()));
        }
    }

    public void sendCustomAudioCaptureAACData(ByteBuffer data, int dataLength, int configLength,
                                              long referenceTimeMillisecond, int samples,
                                              ZGFlutterAudioFrameParam param,
                                              ZGFlutterPublishChannel channel) {
        ZegoAudioFrameParam audioFrameParam = new ZegoAudioFrameParam();
        audioFrameParam.sampleRate = ZegoAudioSampleRate.getZegoAudioSampleRate(param.sampleRate.value());
        audioFrameParam.channel = ZegoAudioChannel.getZegoAudioChannel(param.channel.value());
        if (ZegoExpressEngine.getEngine() != null) {
            ZegoExpressEngine.getEngine().sendCustomAudioCaptureAACData(data, dataLength, configLength, referenceTimeMillisecond, samples, audioFrameParam, ZegoPublishChannel.getZegoPublishChannel(channel.value()));
        }
    }

    public void fetchCustomAudioRenderPCMData(ByteBuffer data, int dataLength,
                                              ZGFlutterAudioFrameParam param) {
        ZegoAudioFrameParam audioFrameParam = new ZegoAudioFrameParam();
        audioFrameParam.sampleRate = ZegoAudioSampleRate.getZegoAudioSampleRate(param.sampleRate.value());
        audioFrameParam.channel = ZegoAudioChannel.getZegoAudioChannel(param.channel.value());
        if (ZegoExpressEngine.getEngine() != null) {
            ZegoExpressEngine.getEngine().fetchCustomAudioRenderPCMData(data, dataLength, audioFrameParam);
        }
    }

    public void sendReferenceAudioPCMData(ByteBuffer data, int dataLength,
                                          ZGFlutterAudioFrameParam param) {
        ZegoAudioFrameParam audioFrameParam = new ZegoAudioFrameParam();
        audioFrameParam.sampleRate = ZegoAudioSampleRate.getZegoAudioSampleRate(param.sampleRate.value());
        audioFrameParam.channel = ZegoAudioChannel.getZegoAudioChannel(param.channel.value());
        if (ZegoExpressEngine.getEngine() != null) {
            ZegoExpressEngine.getEngine().sendReferenceAudioPCMData(data, dataLength, audioFrameParam);
        }
    }
}

File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/windows/ZegoCustomAudioCaptureManager.cpp
#include "include/zego_express_engine/ZegoCustomAudioCaptureManager.h"
#include <ZegoExpressSDK.h>

static std::shared_ptr<ZegoCustomAudioCaptureManager> instance_ = nullptr;
static std::once_flag singletonFlag;

std::shared_ptr<ZegoCustomAudioCaptureManager> ZegoCustomAudioCaptureManager::getInstance() {
    std::call_once(singletonFlag,
                   [&] { instance_ = std::make_shared<ZegoCustomAudioCaptureManager>(); });
    return instance_;
}

void ZegoCustomAudioCaptureManager::sendCustomAudioCaptureAACData(
    unsigned char *data, unsigned int dataLength, unsigned int configLength,
    unsigned long long referenceTimeMillisecond, unsigned int samples,
    ZGFlutterAudioFrameParam param, ZGFlutterPublishChannel channel) {
    ZEGO::EXPRESS::ZegoAudioFrameParam audioFrameParam = ZEGO::EXPRESS::ZegoAudioFrameParam();
    audioFrameParam.channel = ZEGO::EXPRESS::ZegoAudioChannel(param.channel);
    audioFrameParam.sampleRate = ZEGO::EXPRESS::ZegoAudioSampleRate(param.sampleRate);
    if (ZEGO::EXPRESS::ZegoExpressSDK::getEngine()) {
        ZEGO::EXPRESS::ZegoExpressSDK::getEngine()->sendCustomAudioCaptureAACData(
            data, dataLength, configLength, referenceTimeMillisecond, samples, audioFrameParam,
            ZEGO::EXPRESS::ZegoPublishChannel(channel));
    }
}

void ZegoCustomAudioCaptureManager::sendCustomAudioCapturePCMData(unsigned char *data,
                                                                  unsigned int dataLength,
                                                                  ZGFlutterAudioFrameParam param,
                                                                  ZGFlutterPublishChannel channel) {
    ZEGO::EXPRESS::ZegoAudioFrameParam audioFrameParam = ZEGO::EXPRESS::ZegoAudioFrameParam();
    audioFrameParam.channel = ZEGO::EXPRESS::ZegoAudioChannel(param.channel);
    audioFrameParam.sampleRate = ZEGO::EXPRESS::ZegoAudioSampleRate(param.sampleRate);
    if (ZEGO::EXPRESS::ZegoExpressSDK::getEngine()) {
        ZEGO::EXPRESS::ZegoExpressSDK::getEngine()->sendCustomAudioCapturePCMData(
            data, dataLength, audioFrameParam, ZEGO::EXPRESS::ZegoPublishChannel(channel));
    }
}

void ZegoCustomAudioCaptureManager::fetchCustomAudioRenderPCMData(unsigned char *data,
                                                                  unsigned int dataLength,
                                                                  ZGFlutterAudioFrameParam param) {
    ZEGO::EXPRESS::ZegoAudioFrameParam audioFrameParam = ZEGO::EXPRESS::ZegoAudioFrameParam();
    audioFrameParam.channel = ZEGO::EXPRESS::ZegoAudioChannel(param.channel);
    audioFrameParam.sampleRate = ZEGO::EXPRESS::ZegoAudioSampleRate(param.sampleRate);
    if (ZEGO::EXPRESS::ZegoExpressSDK::getEngine()) {
        ZEGO::EXPRESS::ZegoExpressSDK::getEngine()->fetchCustomAudioRenderPCMData(data, dataLength,
                                                                   audioFrameParam);
    }
}

void ZegoCustomAudioCaptureManager::sendReferenceAudioPCMData(unsigned char *data,
                                                              unsigned int dataLength,
                                                              ZGFlutterAudioFrameParam param) {
    ZEGO::EXPRESS::ZegoAudioFrameParam audioFrameParam = ZEGO::EXPRESS::ZegoAudioFrameParam();
    audioFrameParam.channel = ZEGO::EXPRESS::ZegoAudioChannel(param.channel);
    audioFrameParam.sampleRate = ZEGO::EXPRESS::ZegoAudioSampleRate(param.sampleRate);
    if (ZEGO::EXPRESS::ZegoExpressSDK::getEngine()) {
        ZEGO::EXPRESS::ZegoExpressSDK::getEngine()->sendReferenceAudioPCMData(data, dataLength, audioFrameParam);
    }
}